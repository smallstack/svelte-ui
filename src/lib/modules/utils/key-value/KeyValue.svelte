<script lang="ts">
  import type { Snippet } from "svelte"; 

  let { key, value, children }: { key: string; value?: string; children?: Snippet } = $props();
</script>

<div class="flex flex-row place-items-center justify-between">
  <div class="flex flex-col">
    <div class="text font-semibold">{key}</div>
    {#if value !== undefined}
      <div class="text-sm">{value}</div>
    {:else if children}
      {@render children()}
    {:else}
      <div class="text-sm text-base-300">-</div>
    {/if}
  </div>
</div>
