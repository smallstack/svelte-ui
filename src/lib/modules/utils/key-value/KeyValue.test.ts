import { render } from "@testing-library/svelte";
import { describe, expect, test } from "vitest";
import KeyValue from "./KeyValue.svelte";
import KeyValueTest from "./KeyValue.test.svelte";

describe("KeyValue", () => {
	test("renders key correctly", () => {
		const { getByText } = render(KeyValue, { props: { key: "Test Key" } });
		expect(getByText("Test Key")).toBeTruthy();
	});

	test("renders value correctly when provided", () => {
		const { getByText } = render(KeyValue, { props: { key: "Test Key", value: "Test Value" } });
		expect(getByText("Test Value")).toBeTruthy();
	});

	test("renders children correctly when provided", () => {
		const { getByText } = render(KeyValueTest, {});
		expect(getByText("Children")).toBeTruthy();
	});

	test("renders placeholder when neither value nor children are provided", () => {
		const { getByText } = render(KeyValue, { key: "Test Key" });
		expect(getByText("-")).toBeTruthy();
	});

	test("snapshot matches", async () => {
		const { container } = render(KeyValue, { props: { key: "Test Key", value: "Test Value" } });
		expect(container.innerHTML).toMatchSnapshot();
	});
});
