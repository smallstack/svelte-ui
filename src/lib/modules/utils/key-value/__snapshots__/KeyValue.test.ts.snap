// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`KeyValue > snapshot matches 1`] = `"<div class="flex flex-row place-items-center justify-between"><div class="flex flex-col"><div class="text font-semibold">Test Key</div> <div class="text-sm">Test Value</div><!----></div></div>"`;
